### DISCUSSION TITLE
The intended way to disable all fields when form is submitting

### DISCUSSION THREAD

**Discussion by mwskwong** [react-hook-form/react-hook-form#11171] [Q&A]
With the introduction of `disabled` in `useForm`, it is possible to disable the entire form with:

```ts
useForm({ disabled: true })
```

However, the "form is submitting" state seems to be best to be tracked with `formState.isSubmitting` and obviously, this does **NOT** work

```ts
const { formState } = useForm({ disabled: formState.isSubmitting })
```

So am I better off having a separate `disabled` state as illustrated in https://github.com/react-hook-form/react-hook-form/releases/tag/v7.48.0?

Or should I simply set the `disabled` prop of every `Controller`?

```tsx
const { formState } = useForm({ disabled: formState.isSubmitting })

<Controller disabled={formState.isSubmitting} />
<Controller disabled={formState.isSubmitting} />
<Controller disabled={formState.isSubmitting} />
```

**Comment by mwskwong** [comment]
bump
Related question: https://github.com/orgs/react-hook-form/discussions/11057

**Comment by HasanMothaffar** [comment]
+1. Setting the disabled prop in all controllers is unwieldy.

Maybe we could have a field `disabledWhileSubmitting: boolean`?

**Reply by HasanMothaffar** [reply]
@bluebill1049

**Reply by HasanMothaffar-OSG** [reply]
@bluebill1049 It would be helpful to have this field, please
